src/Language/DslTransformer.php

```
@@ -0,0 +1,165 @@
+final class DslTransformer
+{
+    public static function fromDsl(string $dsl, SchemaValidator $validator): AuthorizationModelInterface
+    {
+        $lines = preg_split('/\r?\n/', $dsl);
+        $schemaVersion = SchemaVersion::V1_1;
+        $typeDefinitions = [];
+
+        $currentType = null;
+        $relations = [];
+
+        foreach ($lines as $line) {
```

PHPStan (2.1.15)

Argument of an invalid type list<string>|false supplied for foreach, only iterables are supported.


(foreach.nonIterable)